### PR TITLE
support iscsi task function

### PR DESCRIPTION
**Pull request by carmark**
fix https://github.com/gostor/gotgt/issues/35

### CODE REVIEW

**Comment by carmark** [comment]
@kmova @orzhang 

Finally, I take some weekend to finish this function.  The libiscsi test panic when I did the `iSCSI.iSCSITMF` testsuite, do you have any idea to test this?

File: .travis.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ script:
    - ./autogen.sh
    - ./configure
    - make
+   - ./test-tool/iscsi-test-cu -d -A -V --test=iSCSI.iSCSITMF iscsi://127.0.0.1:3260/${TARGET}/0
```
Comment (gila) [inline .travis.yml]: I can't seem to find this test

Comment (carmark) [inline .travis.yml]:
```
$ ./iscsi-test-cu -l | grep iSCSI.iSCSITMF
iSCSI.iSCSITMF
iSCSI.iSCSITMF.AbortTaskSimpleAsync
```

Comment (gila) [inline .travis.yml]: $git remote -v show
origin	https://github.com/gostor/libiscsi.git (fetch)
origin	https://github.com/gostor/libiscsi.git (push)

$ ./test-tool/iscsi-test-cu -l | grep iSCSI.iSCSITMF

Should be the same repo travis is using.

Comment (carmark) [inline .travis.yml]: I did sync the gostor/libiscsi repo, it should include the iSCSITMF testsuite.

**Comment by carmark** [comment]
After updating the `gostor/libiscsi` repo, the testsuite of `iSCSI.iSCSITMF` has one failed test.  

```
0.20s$ ./test-tool/iscsi-test-cu -d -A -V --test=iSCSI.iSCSITMF iscsi://127.0.0.1:3260/${TARGET}/0
    Send READCAPACITY10 (Expecting SUCCESS) LBA:0 pmi:0
    [OK] READCAPACITY10 returned SUCCESS NO SENSE(0x00) (null)(0x0000)
    Send READCAPACITY16 (Expecting SUCCESS)
    [OK] READCAPACITY16 returned SUCCESS NO SENSE(0x00) (null)(0x0000)
    Send INQUIRY (Expecting SUCCESS) evpd:0 page_code:00 alloc_len:64
    [OK] INQUIRY returned SUCCESS NO SENSE(0x00) (null)(0x0000)
    Send INQUIRY (Expecting SUCCESS) evpd:0 page_code:00 alloc_len:74
    [OK] INQUIRY returned SUCCESS NO SENSE(0x00) (null)(0x0000)
    Send INQUIRY (Expecting SUCCESS) evpd:1 page_code:b0 alloc_len:64
    [FAILED] INQUIRY command failed with sense. ILLEGAL_REQUEST(0x05)/(null)(0x0000)
    Send INQUIRY (Expecting SUCCESS) evpd:1 page_code:b1 alloc_len:255
    [FAILED] INQUIRY command failed with sense. ILLEGAL_REQUEST(0x05)/(null)(0x0000)
Failed to read Block Device Characteristics page
    Send REPORT_SUPPORTED_OPCODE (Expecting SUCCESS) RCTD:1 OPTIONS:0 OPCODE:0x00 SA:0 ALLOC_LEN:65535
    [OK] REPORT_SUPPORTED_OPCODES returned SUCCESS NO SENSE(0x00) (null)(0x0000)
    Send MODESENSE6 (Expecting SUCCESS) 
    [OK] MODESENSE6 returned SUCCESS NO SENSE(0x00) (null)(0x0000)
    Send PRIN/READ_KEYS
    [FAILED] PRIN command: failed with sense. SENSE KEY:ILLEGAL_REQUEST(5) ASCQ:(null)(0x0000)
     CUnit - A unit testing framework for C - Version 2.1-2
     http://cunit.sourceforge.net/
Suite: iSCSITMF
  Test: AbortTaskSimpleAsync ...    WRITE10 queued: (CmdSN=0x1a3c06f2, ITT=0x7a6406f7)
    dispatching out queue...
    dispatched
    ABORT queued: (RefCmdSN=0x1a3c06f2, RefITT=0x7a6406f7)
    dispatching abort and handling responses...
    WRITE10 successful: (CmdSN=0x1a3c06f2, ITT=0x7a6406f7)
FAILED
    1. test_async_abort_simple.c:67  - CU_ASSERT_PTR_NOT_NULL_FATAL(command_data)
  Test: LUNResetSimpleAsync ...    WRITE10 queued: (CmdSN=0x613a3610, ITT=0x14f3839f)
    dispatching out queue...
    dispatched
    LU RESET queued
    dispatching reset and handling responses...
    WRITE10 successful: (CmdSN=0x613a3610, ITT=0x14f3839f)
    LU RESET: TMF response 0 for RefCmdSN=0x613a3610, RefITT=0x14f3839f
    LU RESET completed
    received all expected responses
    1 IOs completed, 1 resets successful
passed
Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites      1      1    n/a      0        0
               tests      2      2      1      1        0
             asserts     41     41     40      1      n/a
Elapsed time =    0.000 seconds
Tests completed with return value: 0
The command "./test-tool/iscsi-test-cu -d -A -V --test=iSCSI.iSCSITMF iscsi://127.0.0.1:3260/${TARGET}/0" exited with 1.
```

**Comment by gila** [comment]
Ok after merge with upstream I know too have the tests, thank you.

**Comment by carmark** [comment]
merge first and will support async command, then un-comment the iSCSI.iSCSITMF test case.

**Pull request merged**